Editor/EditorCore/PolyShapeTool.cs

```diff
@@ -0,0 +1,958 @@
+
+            if(evt.type == EventType.Layout)
+                HandleUtility.AddDefaultControl(m_ControlId);
+
+            if(polygon.polyEditMode == PolyShape.PolyEditMode.Path && !m_PlacingPoint)
+                m_MouseCursor = MouseCursor.ArrowPlus;
+            else if((GUIUtility.hotControl != 0) || m_PlacingPoint)
+                m_MouseCursor = MouseCursor.MoveArrow;
+            else
+                m_MouseCursor = MouseCursor.Arrow;
+
+            DoPointPlacement();
+            DoExistingPointsGUI();
+            DoExistingLinesGUI();
+
+            if(evt.type == EventType.Repaint)
+            {
+                Rect sceneViewRect = window.position;
+                sceneViewRect.x = 0;
+                sceneViewRect.y = 0;
+                SceneView.AddCursorRect(sceneViewRect, m_MouseCursor);
+            }
+        }
+
+        /// <summary>
+        /// Overlay GUI
+        /// </summary>
+        /// <param name="target">the target of this overlay</param>
+        /// <param name="view">the current SceneView where to display the overlay</param>
+        void OnOverlayGUI(UObject target, SceneView view)
+        {
+            switch (polygon.polyEditMode)
+            {
+                case PolyShape.PolyEditMode.Path:
+                {
+                    EditorGUILayout.HelpBox("Click To Add Points\nPress 'Enter' or 'Space' to Set Height", MessageType.Info);
+                    break;
+                }
+
+                case PolyShape.PolyEditMode.Height:
+                {
+                    EditorGUILayout.HelpBox("Move Mouse to Set Height\nPress 'Enter' or 'Space' to Finalize", MessageType.Info);
+                    break;
+                }
+
+                case PolyShape.PolyEditMode.Edit:
+                {
+                    if (GUILayout.Button("Quit Editing", UI.EditorGUIUtility.GetActiveStyle("Button")))
+                        SetPolyEditMode(PolyShape.PolyEditMode.None);
+                    EditorGUILayout.HelpBox("Move Poly Shape points to update the shape\nPress 'Enter' or 'Space' to Finalize", MessageType.Info);
+                    break;
+                }
+            }
+
+            EditorGUI.BeginChangeCheck();
+
+            float extrude = polygon.extrude;
+            extrude = EditorGUILayout.FloatField("Extrusion", extrude);
+
+            bool flipNormals = polygon.flipNormals;
+            flipNormals = EditorGUILayout.Toggle("Flip Normals", flipNormals);
+
+            if (EditorGUI.EndChangeCheck())
+            {
+                if (polygon.polyEditMode == PolyShape.PolyEditMode.None)
+                {
+                    if (ProBuilderEditor.instance != null)
+                        ProBuilderEditor.instance.ClearElementSelection();
+
+                    UndoUtility.RecordObject(polygon, "Change Polygon Shape Settings");
+                    UndoUtility.RecordObject(polygon.mesh, "Change Polygon Shape Settings");
```

Comment:
Could use `UndoUtility.RecordComponents<PolyShape, ProBuilderMesh>(targets as Component[], "Change Polygon Shape Settings");` here

Reply:
I've done that using polygon instead of targets however